### PR TITLE
[feat] implement adjoint for liegroups

### PR DESCRIPTION
**Pull request by cjpurackal**
#### Changes

Implement adjoint representation for liegroups 

#### Type of change

- [ ] 📚  Documentation Update
- [ ] 🧪 Tests Cases
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [x] 🔬 New feature (non-breaking change which adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 This change requires a documentation update

#### Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

File: kornia/geometry/liegroup/so3.py [hunk 1]
```diff
@@ -250,3 +250,15 @@ def inverse(self) -> 'So3':
...
+        """Returns the adjoint matrix of shape :math:`(B, 3, 3)`.
+
+        Example:
+            >>> s = So3.identity()
+            >>> s.adjoint()
+            tensor([[1., 0., 0.],
+                    [0., 1., 0.],
+                    [0., 0., 1.]], grad_fn=<StackBackward0>)
+        """
+        return self.matrix()
```
Comment (edgarriba) [inline kornia/geometry/liegroup/so3.py:264]: a good test here: https://github.com/strasdat/Sophus/blob/23.04-beta/cpp/sophus/lie/so3.h#L145

File: kornia/geometry/liegroup/se3.py [hunk 2]
```diff
@@ -247,3 +247,22 @@ def inverse(self) -> 'Se3':
...
+            tensor([[[1., 0., 0., 0., 0., 0.],
+                     [0., 1., 0., 0., 0., 0.],
+                     [0., 0., 1., 0., 0., 0.],
+                     [0., 0., 0., 1., 0., 0.],
+                     [0., 0., 0., 0., 1., 0.],
+                     [0., 0., 0., 0., 0., 1.]]], grad_fn=<CatBackward0>)
+        """
+        R = self.so3.matrix()
+        z = zeros_like(R)
+        row0 = concatenate((R, self.t * R), -1)
```
Comment (edgarriba) [inline kornia/geometry/liegroup/se3.py:266]:
```suggestion
        row0 = concatenate((R, So3.hat(self.t) * R), -1)
```
https://github.com/strasdat/Sophus/blob/23.04-beta/cpp/sophus/lie/se3.h#L146
and test: https://github.com/strasdat/Sophus/blob/23.04-beta/cpp/sophus/lie/se3.h#L139

File: kornia/geometry/liegroup/se3.py [hunk 3]
```diff
@@ -247,3 +247,22 @@ def inverse(self) -> 'Se3':
...
+
+        Example:
+            >>> s = Se3.identity()
+            >>> s.adjoint()
+            tensor([[[1., 0., 0., 0., 0., 0.],
+                     [0., 1., 0., 0., 0., 0.],
+                     [0., 0., 1., 0., 0., 0.],
+                     [0., 0., 0., 1., 0., 0.],
+                     [0., 0., 0., 0., 1., 0.],
+                     [0., 0., 0., 0., 0., 1.]]], grad_fn=<CatBackward0>)
```
Comment (edgarriba) [inline kornia/geometry/liegroup/se3.py:262]: why it returns batched while in theory you are calling unbatched identity ?

Comment (cjpurackal) [reply kornia/geometry/liegroup/se3.py:262]: It is internally represented as batched

Comment (edgarriba) [reply kornia/geometry/liegroup/se3.py:262]: then not sure it's correct. If Se3 data storage is non-batched, all the chained ops should keep the same data layout

Comment (cjpurackal) [reply kornia/geometry/liegroup/se3.py:262]: Should it return as unbatched for unbatched inputs? Internally representing it as batched reduces a bit of checks, this why i did this way.

Comment (cjpurackal) [reply kornia/geometry/liegroup/se3.py:262]: > then not sure it's correct. If Se3 data storage is non-batched, all the chained ops should keep the same data layout

okay

**Comment by cjpurackal** [comment]
@edgarriba tests for adjoint based on this 
image

File: kornia/geometry/liegroup/se3.py [hunk 4]
```diff
@@ -323,3 +323,22 @@ def trans_z(cls, z: Tensor) -> "Se3":
...
+                    [0., 0., 1., 0., 0., 0.],
+                    [0., 0., 0., 1., 0., 0.],
+                    [0., 0., 0., 0., 1., 0.],
+                    [0., 0., 0., 0., 0., 1.]], grad_fn=<CatBackward0>)
+        """
+        R = self.so3.matrix()
+        z = zeros_like(R)
+        row0 = concatenate((R, So3.hat(self.t) @ R), -1)
+        row1 = concatenate((z, R), -1)
+        return concatenate((row0, row1), 0 if len(R.shape) < 3 else 1)
```
Comment (edgarriba) [inline kornia/geometry/liegroup/se3.py:344]:
```suggestion
        return concatenate((row0, row1), -2)
```

**Review by edgarriba — APPROVED** [review]

**Pull request merged**